packages/contracts/src/utils/JwtVerifier.sol

```diff
@@ -0,0 +1,157 @@
+            COMMAND_BYTES
+        );
+        for (uint256 i = 0; i < COMMAND_FIELDS; i++) {
+            pubSignals[1 + ISS_FIELDS + 3 + i] = stringFields[i];
+        }
+        // accountSalt;
+        pubSignals[1 + ISS_FIELDS + 3 + COMMAND_FIELDS] = uint256(
+            proof.accountSalt
+        );
+        // azp
+        stringFields = _packBytes2Fields(bytes(parts[2]), AZP_BYTES);
+        for (uint256 i = 0; i < AZP_FIELDS; i++) {
+            pubSignals[
+                1 + ISS_FIELDS + 3 + COMMAND_FIELDS + 1 + i
+            ] = stringFields[i];
+        }
+        // isCodeExist;
+        pubSignals[1 + ISS_FIELDS + 3 + COMMAND_FIELDS + 1 + AZP_FIELDS] = proof
+            .isCodeExist
+            ? 1
+            : 0;
+
+        require(
+            groth16Verifier.verifyProof(pA, pB, pC, pubSignals),
+            "Invalid proof"
+        );
+
+        emit JWTProofVerified(proof.maskedCommand);
+    }
+
+    function _packBytes2Fields(
+        bytes memory _bytes,
+        uint256 _paddedSize
+    ) public pure returns (uint256[] memory) {
+        uint256 remain = _paddedSize % 31;
+        uint256 numFields = (_paddedSize - remain) / 31;
+        if (remain > 0) {
+            numFields += 1;
+        }
+        uint256[] memory fields = new uint[](numFields);
+        uint256 idx = 0;
+        uint256 byteVal = 0;
+        for (uint256 i = 0; i < numFields; i++) {
+            for (uint256 j = 0; j < 31; j++) {
+                idx = i * 31 + j;
+                if (idx >= _paddedSize) {
+                    break;
+                }
+                if (idx >= _bytes.length) {
+                    byteVal = 0;
+                } else {
+                    byteVal = uint256(uint8(_bytes[idx]));
+                }
+                if (j == 0) {
+                    fields[i] = byteVal;
+                } else {
+                    fields[i] += (byteVal << (8 * j));
+                }
+            }
+        }
+        return fields;
+    }
+
+    /// @notice Upgrade the implementation of the proxy.
+    /// @param newImplementation Address of the new implementation.
+    function _authorizeUpgrade(
+        address newImplementation
+    ) internal override onlyOwner {}
+
+    function getCommandBytes() external pure returns (uint256) {
+        return COMMAND_BYTES;
+    }
+
+    function stringToArray(
+        string memory _strings
+    ) external pure returns (string[] memory) {
+        strings.slice memory slicee = _strings.toSlice();
+        strings.slice memory delim = "|".toSlice();
+        string[] memory parts = new string[](slicee.count(delim) + 1);
+        for (uint i = 0; i < parts.length; i++) {
+            parts[i] = slicee.split(delim).toString();
+        }
+        require(parts.length == 3, "Invalid kid|iss|azp strings");
+        return parts;
+    }
+
+    function stringToBytes32(
+        string memory source
+    ) public pure returns (bytes32 result) {
```

Comment:
@shreyas-londhe I created the HexUtils.sol, so I moved this function into another contract.

Reply:
If this function is used only in JwtVerifier, I think it is better to just make it public rather than making a new library, which increases complexity especially on zksync.